domain/src/main/java/com/teamagam/gimelgimel/domain/notifications/NotifyOnNewMessageInteractor.java

```diff
@@ -0,0 +1,46 @@
+package com.teamagam.gimelgimel.domain.notifications;
+
+import com.google.auto.factory.AutoFactory;
+import com.google.auto.factory.Provided;
+import com.teamagam.gimelgimel.domain.base.executor.PostExecutionThread;
+import com.teamagam.gimelgimel.domain.base.executor.ThreadExecutor;
+import com.teamagam.gimelgimel.domain.base.interactors.BaseSingleDisplayInteractor;
+import com.teamagam.gimelgimel.domain.base.interactors.DisplaySubscriptionRequest;
+import com.teamagam.gimelgimel.domain.messages.entity.ChatMessage;
+import com.teamagam.gimelgimel.domain.messages.entity.features.AlertFeature;
+import com.teamagam.gimelgimel.domain.messages.repository.MessagesRepository;
+import com.teamagam.gimelgimel.domain.utils.PreferencesUtils;
+
+@AutoFactory
+public class NotifyOnNewMessageInteractor extends BaseSingleDisplayInteractor {
+
+  private final PreferencesUtils mPreferencesUtils;
+  private final MessagesRepository mMessagesRepository;
+  private final NotificationDisplayer mNotificationDisplayer;
+
+  protected NotifyOnNewMessageInteractor(@Provided ThreadExecutor threadExecutor,
+      @Provided PostExecutionThread postExecutionThread,
+      @Provided PreferencesUtils preferencesUtils,
+      @Provided MessagesRepository messagesRepository,
+      NotificationDisplayer notificationDisplayer) {
+    super(threadExecutor, postExecutionThread);
+    mPreferencesUtils = preferencesUtils;
+    mMessagesRepository = messagesRepository;
+    mNotificationDisplayer = notificationDisplayer;
+  }
+
+  @Override
+  protected SubscriptionRequest buildSubscriptionRequest(DisplaySubscriptionRequest.DisplaySubscriptionRequestFactory factory) {
+    return factory.create(mMessagesRepository.getMessagesObservable(),
```

Comment:
It there any way to avoid listening and filtering any incoming chat message?
The current observable is emitting every new item and then filtering it. You can optimize the code by listening only when needed.